### ISSUE TITLE
IntervalScan doesn't work with arrays

### ISSUE THREAD

**Issue by severb** [uber/hyperbahn#279]
I'm using this code to smear a tight loop but it doesn't work on arrays. I think the problem is here: https://github.com/uber/hyperbahn/blob/master/lib/interval-scan.js#L141.

**Comment by kriskowal** [comment]
Correct, this should be an array of indexes, which is *easiest* to obtain with Object.keys(array), proper to generate an array of indexes, better to iterate differently for arrays and objects.
